File: backend/datatypes/result.py
from pydantic import BaseModel
from backend.datatypes.event import EventCreate


class Result(BaseModel):
    resultID: int
    playerID: int
    playerName: str
    teamID: int
    teamName: str
    youtubeURL: str
    gameName: str


class ResultCreate(BaseModel):
    events: list[EventCreate]
    isPlayerMode: bool
    playerID: int
    teamID: int
    youtubeURL: str
    gameName: str


class ResultQuery(BaseModel):
    isPlayerMode: bool
    playerID: int
    teamID: int

File: frontend/src/types/Result.ts
export type Result = {
  resultID: number,
  playerID: number,
  playerName: string,
  teamID: string,
  teamName: string,
  youtubeURL: string,
  gameName: string,
}

export const DEFAULT_RESULT = {
  resultID: 0,
  playerID: 0,
  playerName: '',
  teamID: 0,
  teamName: '',
  youtubeURL: '',
  gameName: '',
}